packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,379 @@
+    require(revocable, "Revoking is not allowed");
+    require(!revoked, "Vesting already revoked");
+    revokeTime = revokeTimestamp > block.timestamp ? revokeTimestamp : block.timestamp;
+    uint256 balance = getGoldToken().balanceOf(address(this));
+    uint256 withdrawableAmount = getWithdrawableAmountAtTimestamp(revokeTime);
+    uint256 refundAmount = balance.sub(withdrawableAmount);
+    revoked = true;
+    require(
+      getGoldToken().transfer(refundDestination, refundAmount),
+      "Transfer of refund upon revokation failed"
+    );
+    emit VestingRevoked(revokeTime);
+  }
+
+  /**
+   * @notice Allows only the revoker to pause the gold withdrawal
+   * @param pausePeriod the period for which the withdrawal shall be paused
+   */
+  function pause(uint256 pausePeriod) external onlyRevoker {
+    require(!paused, "Vesting withdrawals already paused");
+    require(revocable, "Vesting must be revokable");
+    require(!revoked, "Vesting already revoked");
+    require(pausePeriod <= 365 days, "Pause period is limited to max. 365 days");
+    paused = true;
+    pauseEndTime = block.timestamp.add(pausePeriod);
+    emit WithdrawalPaused(pausePeriod, block.timestamp);
+  }
+
+  /**
+   * @dev Calculates the amount that has already vested but hasn't been withdrawn yet.
+   * @param timestamp the timestamp at which the calculate the withdrawable amount
+   * @return The withdrawable amount at the timestamp
+   * @dev Function is also made public in order to be called for informational purpose
+   */
+  function getWithdrawableAmountAtTimestamp(uint256 timestamp) public view returns (uint256) {
+    return calculateFreeAmountAtTimestamp(timestamp).sub(currentlyWithdrawn);
+  }
+
+  /**
+   * @dev Calculates the amount that has already vested.
+   * @param timestamp the timestamp at which the calculate the already vested amount
+   */
+  function calculateFreeAmountAtTimestamp(uint256 timestamp) private view returns (uint256) {
+    uint256 currentBalance = getGoldToken().balanceOf(address(this));
+    uint256 totalBalance = currentBalance.add(currentlyWithdrawn);
+
+    if (timestamp < vestingScheme.vestingCliffStartTime) {
+      return 0;
+    }
+    if (
+      timestamp >=
+      vestingScheme.vestingStartTime.add(
+        vestingScheme.vestingPeriods.mul(vestingScheme.vestingPeriodSec)
+      ) ||
+      revoked
+    ) {
+      return totalBalance;
+    }
+
+    uint256 vestingCurveGradient = (timestamp.sub(vestingScheme.vestingStartTime)).div(
+      vestingScheme.vestingPeriods
+    );
+    return (totalBalance.mul(vestingCurveGradient)).div(vestingScheme.vestingPeriodSec);
+  }
+
+  /**
+   * @notice A wrapper func for the lock gold method
+   * @param value the value of gold to be locked
+   * @dev To be called only by the beneficiary of the vesting
+   */
+  function lockGold(uint256 value) external nonReentrant onlyBeneficiary {
+    // the beneficiary may not lock more than the vesting currently has available
```

Comment:
Nit: comment might be unnecessary given the `require` string explains the same thing.